Add section on overflow (#66)
mpusz/wg21-papers: src/3045R5_quantities_and_units_library.md

| SUPERSCRIPT MINUS     |   ⁻    | u8"\u207b" |         "-"          |
| MULTIPLICATION SIGN   |   ×    | u8"\u00d7" |         "x"          |
| GREEK SMALL LETTER PI |   π    | u8"\u03c0" |         "pi"         |
| DOT OPERATOR          |   ⋅    | u8"\u22C5" |      `<none>`[^1]    |

[^1]: Users should not select `unit_symbol_separator::half_high_dot` and `character_set::portable`
at the same time. This symbol is valid only for UTF-8 encoding. Otherwise, we propose to throw

> an exception during the unit symbol string processing._

Reviewer: ```suggestion
an exception during the unit symbol string processing.
```

| GREEK SMALL LETTER PI |   π    | u8"\u03c0" |         "pi"         |
| DOT OPERATOR          |   ⋅    | u8"\u22C5" |      `<none>`*       |

> _NOTE: Users should not select `unit_symbol_separator::half_high_dot` and `character_set::portable`

Reviewer: This star referred to line 3992, which had this star as well. Do you have a better idea of how to connect those?

Author: I didn't know what the star was there for, but this makes sense.  I switched to the same footnote notation I was using elsewhere.  I verified that it works in the rendered output.

5. `std::chrono::duration` uses 'Q' and 'q' for a number and a unit. In the grammar above, we
   proposed using 'N' and 'U' for them, respectively. We also introduced 'D' for dimensions. Are
   we OK with this?

> 6. Are we OK with the usage of `_` for denoting a subscript identifier? Should we use it everywhere

Reviewer: With single quotes, I wanted to express that I mean a character, as was the case for `Q` and `q` in the previous point. Maybe we should use `'_'` instead everywhere?

Author: Done.

hurt generic programming.
As such, users should instead use safer representation types.

### Integer overflow

> To convert a quantity in a program to different units, we need to multiply or divide by a conversion
> factor.  Sometimes, the result is too big to fit in the type: a problem known as _overflow_.  While

Reviewer: ```suggestion
To convert a quantity to different units, we must multiply or divide its numerical value by a conversion
factor.  Sometimes, the result is too big to fit in the type: a problem known as _overflow_.  While
```

any numerical type can overflow, we focus on integral types here, because they carry the most risk.
Even the smallest floating point type in common use, `float`, has a range of $10^{38}$, while the
diameter of the observable universe measured in atomic diameters is "only" about $10^{37}$![^333]

> [^333]: Here, we take the radius of the observable universe as 46.6 billion light years, and the

Reviewer: Why do you use `^333` here and `^2` below? Did you mean `^1` here?

Author: Hah... that was just me being lazy.  :sweat_smile:  I didn't know if we had any footnotes before, but I was pretty sure we wouldn't have 333 of them!  I cleaned up the footnote numbering in the latest commits.
